### ISSUE TITLE
`repy.py` option parser consumes arguments meant for the sandboxed program

### ISSUE THREAD

**Issue by aaaaalbert** [SeattleTestbed/repy_v2#101]
@yyzhuang reports that the RepyV2 sandbox consumes (and gets confused by) command-line arguments meant for the sandboxed program:
```
python repy.py restrictionsfile my_program.r2py --my-arg value
Usage: repy.py [options] resource_file program_to_run.r2py [program args]

repy.py: error: no such option: --my-arg
```
Clearly, `repy.py` should just pass `["--my-arg", "value"]` as the `callargs` to the sandboxed program, and not try interpreting them itself.

When running on a remote vessel, the above error message becomes a somewhat puzzling
```
me@target !> start  my_program.r2py --my-arg value
Node Manager warning 'Did not start in a timely manner'
An error occurred: Node Manager warning 'Did not start in a timely manner'
Affected vessels: 7bfaf92a3257a9a3ae59075d45557d87a7379fde:1224:v1.
```

The problem lies in RepyV2's use of the `optparse` Python module. Unlike `getopt` (which RepyV1 used), it needs explicit setup to stop interpreting command-line args as things it should parse via the `disable_interspersed_args()` call (see the [Python `optparse` docs](https://docs.python.org/2/library/optparse.html#querying-and-manipulating-your-option-parser) for details.

**Comment by yyzhuang** [comment]
I pulled repy.py from my phone (the version with json enabled) and added the line `parser.disable_interspersed_args()`, pushed it back. However, I still got the same error:

`%1 !> start dylink.r2py argtest.r2py -s 10000`
`Node Manager warning 'Did not start in a timely manner'`
`An error occurred: Node Manager warning 'Did not start in a timely manner'`
`Affected vessels: 7bfaf92a3257a9a3ae59075d45557d87a7379fde:1224:v1.`

However the old trick works:
`%1 !> start dylink.r2py argtest.r2py -- -s 10000`
`%1 !> show log`
`seconds!!!: 10000`

To make sure the repy code is updated:
`$ cat repy.py`
......

  `### PARSE OPTIONS.   These are command line in our case, but could be from`
  `### anywhere if this is repurposed...`
  `usage = "USAGE: repy.py [options] resource_file program_to_run.r2py [program args]"`
  `parser = optparse.OptionParser(usage=usage)`
  `parser.disable_interspersed_args()`
  `add_repy_options(parser)`
  `options, args = parser.parse_args()`

Is it possible that somewhere else has used optparse?

**Comment by aaaaalbert** [comment]
Oh, you need to update `repy.py` in two places actually: Once in `seattle/seattle_repy`, and then again in `seattle/seattle_repy/repyV2`. The former is used by the nodemanager etc., the latter is the actual sandbox. Sorry for not thinking about this.

**Comment by yyzhuang** [comment]
Yay I think it now works!!

`%1 !> start dylink.r2py argtest.r2py -s 10000`
`%1 !> show log`
`seconds!!!: 10000`

**Comment by aaaaalbert** [comment]
Champagne!

**Comment by vladimir-v-diaz** [comment]
You say this is a quick fix.  Is this change temporary, or will this module eventually switch to using `argparse`?  optparse has been deprecated:

"Deprecated since version 2.7: The optparse module is deprecated and will not be developed further; development will continue with the argparse module."

https://docs.python.org/2/library/optparse.html#querying-and-manipulating-your-option-parser

**Comment by aaaaalbert** [comment]
I called #102 a "quick fix" because it doesn't replace `optparse`. Once merged, we'll have a follow-up issue documenting the need of refactoring (per your comment).

**Comment by aaaaalbert** [comment]
Fixed by #102.
